src/puppetlabs/puppetdb/scf/migrate.clj

```diff
@@ -1491,28 +1495,49 @@
     (into (sorted-map)
           (select-keys migrations pending))))
 
+(defn- sql-or-die [f]
+  "Calls (f) and returns its result.  If f throws an SQLException,
+  logs the exception and its parent and then calls (System/exit 1)"
+  ;; Here we've preserved existing behavior, but this may warrant
+  ;; further consideration later.  If possible, we might want to
+  ;; avoid System/exit in favor of careful exception
+  ;; handling (cf. PDB-1118).
+  (try
+    (f)
+    (catch java.sql.SQLException e
+      (log/error e "Caught SQLException during migration")
+      (let [next (.getNextException e)]
+        (when-not (nil? next)
+          (log/error next "Unravelled exception")))
+      (binding [*out* *err*] (flush)) (flush)
+      (System/exit 1))))
```

Comment:
Could we call the trapperkeeper shutdown function? Am I mistaken or do we not do that elsewhere